models/repo_collaboration.go

```diff
@@ -42,7 +42,7 @@ func (repo *Repository) AddCollaborator(u *User) error {
 	}
 
 	if repo.Owner.IsOrganization() {
-		err = repo.recalculateTeamAccesses(sess, 0)
+		err = repo.recalculateUserAccess(sess, u.ID)
 	} else {
 		err = repo.recalculateAccesses(sess)
```

Comment:
Why not simply use recalculateUserAccess() in both cases here?

Reply:
I handle user repositories now as well.